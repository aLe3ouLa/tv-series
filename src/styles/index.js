import React from "react";

import { Reset, Typography } from "./defaults";
import { color } from "./variables/colors";
import { spacing } from "./variables/spacings";
import { breakpoint } from "./variables/breakpoints";

export const Defaults = () => (
  <>
    <Reset />
    <Typography />
  </>
);

export const theme = {
  color,
  spacing,
  breakpoint,
};
