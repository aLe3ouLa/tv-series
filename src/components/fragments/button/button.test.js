import React from "react";
import Button from "./";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../styles";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Button component", () => {
  const mockProps = {
    onClick: jest.fn(),
  };
  afterEach(cleanup);


  it("section have a button element", async () => {
    const { getByRole } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Button {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const button = getByRole("button");
    expect(button).toBeInTheDocument();
  });

  it("should be able to clicked", async () => {
    const { getByRole } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Button {...mockProps} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const button = getByRole("button");
    fireEvent.click(button);
  });
});
