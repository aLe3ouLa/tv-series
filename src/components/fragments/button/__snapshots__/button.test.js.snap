// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render button component without crashing 1`] = `
<div>
  <button
    class="sc-AxjAm fWcGgS"
  />
</div>
`;
